addons/udes_stock/models/product_product.py

```diff
@@ -81,17 +79,47 @@ def get_product(self, product_identifier):
         """ Get product from a name, barcode, or id.
         """
         if isinstance(product_identifier, int):
-            domain = [('id', '=', product_identifier)]
+            domain = [("id", "=", product_identifier)]
         elif isinstance(product_identifier, str):
-            domain = ['|', ('barcode', '=', product_identifier),
-                           ('name', '=', product_identifier)]
+            domain = ["|", ("barcode", "=", product_identifier), ("name", "=", product_identifier)]
         else:
-            raise ValidationError(_('Unable to create domain for product search from identifier of type %s') % type(product_identifier))
+            raise ValidationError(
+                _("Unable to create domain for product search from identifier of type %s")
+                % type(product_identifier)
+            )
 
         results = self.search(domain)
         if not results:
-            raise ValidationError(_('Invalid product scanned: %s') % str(product_identifier))
-        if  len(results) > 1:
-            raise ValidationError(_('Too many products found for identifier %s') % str(product_identifier))
+            raise ValidationError(_("Invalid product scanned: %s") % str(product_identifier))
+        if len(results) > 1:
+            raise ValidationError(
+                _("Too many products found for identifier %s") % str(product_identifier)
+            )
 
         return results
+
+    @api.multi
+    def sync_active_to_templates(self, activate_templates=True, deactivate_templates=True):
+        """ Copies the active state from products to their templates
+        """
+        ProductTemplate = self.env["product.template"]
+
+        # Prefetch fields
+        self = self.with_context(prefetch_fields=False)
+        self.mapped("active")
+        self.mapped("product_tmpl_id.active")
+
+        # Gather lists of product templates to activate and deactivate
+        templates_to_activate = ProductTemplate.browse()
+        templates_to_deactivate = ProductTemplate.browse()
+        for product in self:
+            if product.active != product.product_tmpl_id.active:
+                if product.active:
+                    templates_to_activate |= product.product_tmpl_id
+                else:
+                    templates_to_deactivate |= product.product_tmpl_id
+
+        if activate_templates:
+            templates_to_activate.write({"active": True})
+        if deactivate_templates:
```

Comment:
I would deactivate first, just in case we ever get in a point where there is no one2one relation.

Reply:
Updated the code